lib/ecto/association.ex

```diff
@@ -176,10 +176,14 @@ defmodule Ecto.Association do
   end
 
   def assoc_query(refl, t, query, values) do
-    query = query || %Ecto.Query{
-      from: %FromExpr{source: {"join expression", nil}},
-      prefix: refl.queryable.__schema__(:prefix)
-    }
+    query =
+      query ||
+      %Ecto.Query{
+        from: %FromExpr{
+          source: {"join expression", nil},
+          prefix: refl.queryable.__schema__(:prefix)
+        }
+      }
```

Comment:
User backwards incompatible!!!

Reply:
Maybe `static_prefix` for a prefix that should always apply for a given schema (unless overridden on-site)?